daemon/checkpoint.go

```diff
@@ -8,6 +8,12 @@ import (
 	"path/filepath"
 
 	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/utils"
+)
+
+var (
+	validCheckpointNameChars   = utils.RestrictedNameChars
+	validCheckpointNamePattern = utils.RestrictedNamePattern
```

Comment:
Thanks @yongtang! Don't do those changes as part of this PR, because I know there were some difficulties last time someone looked at it, so best keep it separate :heart: